client/forest-manager/src/prove.rs

```diff
@@ -23,143 +27,67 @@ use crate::{traits::ForestStorage, types::ForestStorageErrors, utils::deserializ
 /// # Errors
 /// This function can return an error in cases where it fails to read or seek within the trie,
 /// or when deserialization of a leaf's value fails.
-pub(crate) fn prove<T: TrieLayout, F: ForestStorage>(
+pub(crate) fn prove<T: TrieLayout>(
     trie: &trie_db::TrieDB<'_, '_, T>,
-    challenged_file_key: &F::LookupKey,
-) -> Result<Proven<F::RawKey, F::Value>, ForestStorageErrors> {
+    challenged_file_key: &<T::Hash as Hasher>::Out,
+) -> Result<Proven<HasherOutT<T>, ()>, ErrorT<T>>
```

Comment:
The value would be the `Metadata` here, right? Why are you not returning it anymore? Wouldn't it be useful for whoever is trying to prove something? And it's not that much of an added effort.

Reply:
We don't need to insert the metadata as a value of the file key leaf since the only role that forest storage has to fulfilling is to track and prove file key inclusion. When a BSP requires metadata of a file, they will get it from the file storage. It also reduces the storage capacity taken by the forest storage and avoids redundant data being stored in both file and forest storage.

Reply:
Not needed to have metadata here, it would be duplicate information, and with just the file keys gotten from this function, a task could fetch the metadata for those file keys from the file-manager client